src/campaigns/tasks/services/aiCampaignManagerIntegration.service.ts

```diff
@@ -0,0 +1,416 @@
+import { Injectable, Logger } from '@nestjs/common'
+import { Campaign, PathToVictory } from '@prisma/client'
+import {
+  AiCampaignManagerService,
+  StartCampaignPlanRequest,
+} from './aiCampaignManager.service'
+import { CampaignTask, CampaignTaskType } from '../campaignTasks.types'
+
+type CampaignWithPathToVictory = Campaign & {
+  pathToVictory?: PathToVictory | null
+}
+
+@Injectable()
+export class AiCampaignManagerIntegrationService {
+  private readonly logger = new Logger(AiCampaignManagerIntegrationService.name)
+
+  constructor(private readonly aiCampaignManager: AiCampaignManagerService) {}
+
+  async generateCampaignTasks(
+    campaign: CampaignWithPathToVictory,
+  ): Promise<CampaignTask[]> {
+    try {
+      const request = this.buildCampaignPlanRequest(campaign)
+
+      const session =
+        await this.aiCampaignManager.startCampaignPlanGeneration(request)
+      this.logger.log(
+        `Started campaign plan generation with session ID: ${session.session_id}`,
+      )
+
+      await this.aiCampaignManager.waitForCompletion(session.session_id)
+      this.logger.log(
+        `Campaign plan generation completed for session: ${session.session_id}`,
+      )
+
+      const campaignPlanJson = await this.aiCampaignManager.downloadJson(
+        session.session_id,
+      )
+
+      return this.parseCampaignPlanToTasks(campaignPlanJson, campaign)
+    } catch (error) {
+      this.logger.error('Failed to generate campaign tasks', error)
+      throw error
+    }
+  }
+
+  private buildCampaignPlanRequest(
+    campaign: CampaignWithPathToVictory,
+  ): StartCampaignPlanRequest {
+    const { details, data, pathToVictory } = campaign
+
+    const office = details.office || details.normalizedOffice || 'Local Office'
+    const jurisdiction = details.state || 'Unknown State'
+    const district = details.district ? ` - ${details.district}` : ''
+    const office_and_jurisdiction = `${office} in ${jurisdiction}${district}`
+
+    const pathData = pathToVictory?.data as Record<string, unknown> | undefined
```

Comment:
Please type this.. we _know_ what `PathToVictory.data` _is_.. we even have a type for it already.